cupy/linalg/_util.py

```diff
@@ -151,3 +151,19 @@ def _tril(x, k=0):
 def _triu(x, k=0):
     _triu_kernel(k, x)
     return x
+
+
+def batch_identity(batch_shape, n, dtype):
+    shape = batch_shape + (n, n)
+    idx = cupy.arange(n)
+    x = cupy.zeros(shape, dtype)
+    x[..., idx, idx] = 1
+    return x
+
+
+def batch_identity_like(x):
```

Comment:
Should we make sure `x` is a (batch of) square matrices here?

Reply:
Added a comment to the code.